libflux/src/semantic/tests.rs

```diff
@@ -255,715 +255,2377 @@ fn literals() {
+    }
+    test_infer_err! {
+        env: map![
+            "a" => "forall [] string",
+            "b" => "forall [] int",
+            "c" => "forall [] int",
+        ],
+        src: r#"
+            d = if a then b else c
+        "#,
+    }
+    test_infer_err! {
+        env: map![
+            "a" => "forall [] duration",
+            "b" => "forall [] int",
+            "c" => "forall [] int",
+        ],
+        src: r#"
+            d = if a then b else c
+        "#,
+    }
+    test_infer_err! {
+        env: map![
+            "a" => "forall [] time",
+            "b" => "forall [] int",
+            "c" => "forall [] int",
+        ],
+        src: r#"
+            d = if a then b else c
+        "#,
+    }
+    test_infer_err! {
+        env: map![
+            "a" => "forall [] regexp",
+            "b" => "forall [] int",
+            "c" => "forall [] int",
+        ],
+        src: r#"
+            d = if a then b else c
+        "#,
+    }
+    test_infer_err! {
+        env: map![
+            "a" => "forall [] [int]",
+            "b" => "forall [] int",
+            "c" => "forall [] int",
+        ],
+        src: r#"
+            d = if a then b else c
+        "#,
+    }
+    test_infer_err! {
+        env: map![
+            "a" => "forall [] {a: int | b: [float]}",
+            "b" => "forall [] int",
+            "c" => "forall [] int",
+        ],
+        src: r#"
+            d = if a then b else c
+        "#,
+    }
+    test_infer_err! {
+        env: map![
+            "a" => "forall [] bool",
+            "b" => "forall [] int",
+            "c" => "forall [] float",
+        ],
+        src: r#"
+            d = if a then b else c
+        "#,
+    }
+    // test_infer! {
+    //     env: map![
+    //         "a" => "forall [] t0",
+    //         "b" => "forall [] t1",
+    //         "c" => "forall [] t2",
+    //     ],
+    //     src: r#"
+    //            d = if a then b else c
+    //        "#,
+    //     exp: map![
+    //         "a" => "forall [] bool",
+    //         "b" => "forall [] t3",
+    //         "c" => "forall [] t3",
+    //         "d" => "forall [] t3",
+    //     ],
+    // }
+    // test_infer! {
+    //     env: map![
+    //         "a" => "forall [] t0",
+    //         "b" => "forall [] float",
+    //         "c" => "forall [] float",
+    //     ],
+    //     src: r#"
+    //            d = if a then b else c
+    //        "#,
+    //     exp: map![
+    //         "a" => "forall [] bool",
+    //         "d" => "forall [] float",
+    //     ],
+    // }
+    // test_infer! {
+    //     env: map![
+    //         "a" => "forall [] t0",
+    //         "b" => "forall [] t1",
+    //         "c" => "forall [] float",
+    //     ],
+    //     src: r#"
+    //            d = if a then b else c
+    //        "#,
+    //     exp: map![
+    //         "a" => "forall [] bool",
+    //         "b" => "forall [] float",
+    //         "d" => "forall [] float",
```

Comment:
why are tests with `TVar` skipped?

Reply:
We will test tvar instantiation indirectly using function expressions. The plan is to refactor these commented out tests once we can type function expressions.

Reply:
I see, we need unification first.